modules/tensor_mechanics/src/materials/RecomputeRadialReturnPowerLawCreep.C

```diff
@@ -0,0 +1,94 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RecomputeRadialReturnPowerLawCreep.h"
+
+#include "Function.h"
+
+template<>
+InputParameters validParams<RecomputeRadialReturnPowerLawCreep>()
+{
+  InputParameters params = validParams<RecomputeRadialReturn>();
+
+  // Linear strain hardening parameters
+  params.addRequiredParam<Real>("coefficient", "Leading coefficent in power-law equation");
+  params.addRequiredParam<Real>("n_exponent", "Exponent on effective stress in power-law equation");
+  params.addParam<Real>("m_exponent", 0.0, "Exponent on time in power-law equation");
+  params.addRequiredParam<Real>("activation_energy", "Activation energy");
+  params.addParam<Real>("gas_constant", 8.3143, "Universal gas constant");
+  params.addParam<Real>("start_time", 0.0, "Start time (if not zero)");
+  params.addCoupledVar("temp", 0.0, "Coupled temperature");
+
+  return params;
+}
+
+
+RecomputeRadialReturnPowerLawCreep::RecomputeRadialReturnPowerLawCreep(const InputParameters & parameters) :
+    RecomputeRadialReturn(parameters),
+    _coefficient(parameters.get<Real>("coefficient")),
+    _n_exponent(parameters.get<Real>("n_exponent")),
+    _m_exponent(parameters.get<Real>("m_exponent")),
+    _activation_energy(parameters.get<Real>("activation_energy")),
+    _gas_constant(parameters.get<Real>("gas_constant")),
+    _start_time(getParam<Real>("start_time")),
+    _shear_modulus(0.0),
+    _has_temp(isCoupled("temp")),
+    _temperature(_has_temp ? coupledValue("temp") : _zero),
+    _creep_strain(declareProperty<RankTwoTensor>(_base_name + "creep_strain")),
+    _creep_strain_old(declarePropertyOld<RankTwoTensor>(_base_name + "creep_strain"))
+{
+}
+
+void
+RecomputeRadialReturnPowerLawCreep::initQpStatefulProperties()
+{
+  _creep_strain[_qp].zero();
+  _creep_strain_old[_qp].zero();
+}
+
+void
+RecomputeRadialReturnPowerLawCreep::resetQpProperties()
+{
+  _creep_strain[_qp] = _creep_strain_old[_qp];
+
+  RecomputeRadialReturn::resetQpProperties();
+}
+
+void
+RecomputeRadialReturnPowerLawCreep::computeStressInitialize(Real effectiveTrialStress)
+{
+  _shear_modulus = getIsotropicShearModulus();
+
+  _exponential = 1;
+  if (_has_temp)
+  {
+    _exponential = std::exp(-_activation_energy / (_gas_constant * _temperature[_qp]));
+  }
```

Comment:
If you push changes anyways... Single statement if bodies should not have curly braces. And I think an `if` `else` would be a bit more expressive.

```
+  _exponential = 1;
  if (_has_temp)
    _exponential = std::exp(-_activation_energy / (_gas_constant * _temperature[_qp]));
  else
    _exponential = 1.0;
```